daemon/images/image_delete.go

```diff
@@ -73,7 +77,25 @@ func (i *ImageService) ImageDelete(imageRef string, force, prune bool) ([]types.
 	}
 
 	imgID := img.ID()
+
 	repoRefs := i.referenceStore.References(imgID.Digest())
+	defer func() {
+		// We only want to delete the lease if this is the last reference to this image
```

Comment:
Wouldn't it make sense to connect this with the imageStore level and delete lease on `imageStore.Delete()`.

Reply:
Good call, much simpler.